test/unit/test-shadow-embed.js

```diff
@@ -316,6 +318,63 @@ describes.sandboxed('shadow-embed', {}, () => {
     });
   });
 
+  describe('installShadowStyle', () => {
+    let shadowRoot, shadowRoot2;
+
+    beforeEach(() => {
+      shadowRoot = document.createElement('div');
+      shadowRoot2 = document.createElement('div');
+    });
+
+    afterEach(() => {
+      resetShadowStyleCacheForTesting(window);
+    });
+
+    describe('adopted stylesheets supported', () => {
+      before(function () {
+        if (!document.adoptedStyleSheets) {
+          this.skipTest();
```

Comment:
Note: I'm not seeing this test being skipped when running `gulp unit --files=test/unit/test-shadow-embed.js`. Is there an environment where this is expected?

Reply:
Another testing note: looks like this breaks `extensions/amp-base-carousel/1.0/test/test-amp-base-carousel.js`:
```
Error: Uncaught Error: NotAllowedError: Failed to set the 'adoptedStyleSheets' property on 'ShadowRoot': Sharing constructed stylesheets in multiple documents is not allowed
```

Reply:
^ It appears to be a mismatch between the testing `env.win.document` and the global `document`

Reply:
It wouldn't work in Safari at this time. Thanks for the note on the test. I'll look into that.

Reply:
Ok. The error itself turned out to be very simple. I just needed to use `new win.CSSStyleSheet()` vs `new CSSStyleSheet()`. The bottom line: our tests work!